### ISSUE TITLE
Bug/request: Support ssh git sources

### ISSUE THREAD

**Issue by GeorgeHahn** [EmbarkStudios/cargo-deny#148]
**Describe the bug**
Crates fetched via SSH don't appear to be supported for source checking. I'm not sure if this is a bug or a feature request, but it would be nice for projects that want to deny unknown sources but fetch some dependencies from git with ssh.

**To Reproduce**

In `deny.toml`:
```
[sources]
unknown-git = "deny"
allow-git = [
    "ssh://github.com/organization/project",
]
```

`check sources` fails with this error:
```
error: detected 'git' source not specifically allowed

    ┌── /home/user/project/Cargo.lock:37:27 ───
    │
 37 │ project 0.1.0 git+ssh://git@github.com/organization/project.git#<commit>
    │                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source
```

**Expected behavior**
Check succeeds

**Additional context**

Related to #115

**Comment by senden9** [comment]
I am not able to reproduce that error. See the attached shell session & toml files for reference.
```
➜  testing git:(master) ✗ cargo deny check sources
sources ok
➜  testing git:(master) ✗ cargo deny --version
cargo-deny 0.7.3
```
`Cargo.toml`:
```toml
[package]
name = "testing"
version = "0.1.0"
authors = ["Stefano Probst <senden9@gmail.com>"]
edition = "2018"
license = "MIT"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
cargo-deny = { git = "ssh://git@github.com/EmbarkStudios/cargo-deny.git", tag="0.7.3" }
```
`deny.toml`:
```toml
[sources]
# Lint level for what to happen when a crate from a crate registry that is not
# in the allow list is encountered
unknown-registry = "deny"
# Lint level for what to happen when a crate from a git repository that is not
# in the allow list is encountered
unknown-git = "deny"
# List of URLs for allowed crate registries. Defaults to the crates.io index
# if not specified. If it is specified but empty, no registries are allowed.
allow-registry = ["https://github.com/rust-lang/crates.io-index"]
# List of URLs for allowed Git repositories
allow-git = ["ssh://git@github.com/EmbarkStudios/cargo-deny.git"]
```

Have I done something other so that it works now or has this bug solved itself in the meantime?

**Comment by Jake-Shadle** [comment]
This _should_ be resolved by using the git cli support added in #420
